src/App.jsx

```diff
@@ -1,22 +1,32 @@
-import React, { useState } from "react";
-import TodoList from "./TodoList";
-import AddTodoForm from "./AddTodoForm";
- 
+   
+   // eslint-disable-next-line no-unused-vars
+   import React, { useEffect, useState } from "react";
+    import TodoList from "./TodoList";
+    import AddTodoForm from "./AddTodoForm";
+        function useSemiPersistentState (){
+        const [todoList, setTodoList] = useState(() => {
+        const savedTodoList = localStorage.getItem ('savedTodoList');
+            return  savedTodoList ? JSON.parse (savedTodoList) : [];
+           });
+  useEffect(() => {
+    localStorage.setItem ('savedTodoList',JSON.stringify(todoList))},[todoList]);
+   return [todoList,setTodoList];
+}
```

Comment:
Try to maintain proper formatting of whitespace. Make sure your indentation make sense 👍